### DISCUSSION TITLE
Websocket: one connection, multiple payloads

### DISCUSSION THREAD

**Discussion by lukepighetti** [Kong/insomnia#5911] [Ideas]
I have noticed that very few WebSocket server implementations expect someone to open one websocket connection for every type of message they want to send.

Instead, what is more popular, is opening a single WebSocket connection and then dispatching multiple different events to interact with a service.

This is currently hard to do in Insomnia.

What would be interesting to me, is having a folder-like WebSocket, which can contain multiple different payloads which can be run under it. The connection indicator would be on the top level WebSocket, not on the individual payloads

**Comment by HughMacdonald** [comment]
I'm currently in the process of trying to switch over from Postman to Insomnia for a project, and this is going to be a blocker for me. I have a single WebSocket connection and I need to be able to send multiple messages through it

**Reply by john-rager** [reply]
This is a huge blocker for me, as this is the WHOLE point of Websockets. HTTP is stateless so every request must include all information necessary to connect and perform one single request. Websockets establishes a persistent connection: connect, send/receive messages, disconnect.

I gave Insomnia a try because the interface appeared much less cluttered than Postman. But it can't work like this. Please don't advertise that you support Websockets when it doesn't support the most basic functionality of a Websocket.

On top of this, despite the cleaner interface, it has many glitches. App wouldn't start after I signed-up - got a very generic message "failed to launch" and to contact support after a couple of seconds. Displayed on top of the "manual" launch button, so had to hurry and click the button before the error appeared.

Also, rendering of the events panel displayed some error about not being able to render toString, and wouldn't render any more until I restarted the app.

Not bothering to report these issues to support. For v10, this is definitely not a production quality app. These are issues that should have been addressed in alpha/beta. Uninstalling, and highly doubt I'll be back.
